### PR TITLE
Update nats-py lib + Code refactor + Black reformat

### PR DESCRIPTION
**Pull request by ixjosemi**
Hi team,

## Content
- Updated the `nats-py` library from `v0.10.0` to `2.1.0`
- Refactorized the kre-entrypoint and kre-py code to adapt it to the lib update.
- Executed black to reformat all the python code (`$ black --line-length=100`)

### Dev environment successfully deployed
image

image

### Version successfully running
image

### Message successfully sent and received
image

image

### Stopping the version
image

Best regards,
JM.

### CODE REVIEW

File: .github/workflows/kre-py_tests.yaml [hunk 1]
```diff
@@ -39,6 +39,7 @@ jobs:
     - name: Install dependencies
       run: |
         pipenv install --deploy --dev
+        sleep 120
```
Comment (alejandrogr) [inline .github/workflows/kre-py_tests.yaml:42]: When addind a sleep to the code, please add a comment on why we need it

Comment (blopezpi) [reply .github/workflows/kre-py_tests.yaml:43]: Could you check the comment in the workflow? I think that this is the principal reason, but I would like to be sure about that.

Comment (alejandrogr) [reply .github/workflows/kre-py_tests.yaml:43]: LGTM!

**Comment by alejandrogr** [comment]
@ixjosemi just a comment for the next time: code style refactors MUST be in a separate commit :D

**Comment by alejandrogr** [comment]
@ixjosemi Sonar un failing due to three new code smells introduced in this PR. I've checked them and they seem to be there before and not introduced in this PR.

However we can fix them: just create a custom exception for the prediction error instead of raising a generic Exception.

**Pull request closed without merging**